core/src/test/java/org/elasticsearch/index/query/SimpleQueryStringBuilderTest.java

```diff
@@ -0,0 +1,358 @@
+    }
+
+    @Test
+    public void testDefaultEmptyComplainFlags() {
+        qb.flags(new SimpleQueryStringFlag[]{});
+        assertNotNull("Setting flags to empty array should result in query validation exception.", qb.validate());
+    }
+
+    @Test
+    public void testDefaultNullComplainOp() {
+        qb.defaultOperator(null);
+        assertNotNull("Setting default operator to null should result in query validation exception.", qb.validate());
+    }
+
+    // Check operator handling, and default field handling.
+    @Test
+    public void testDefaultOperatorHandling() {
+        SimpleQueryStringBuilder qb = new SimpleQueryStringBuilder("The quick brown fox.");
+        BooleanQuery boolQuery = (BooleanQuery) qb.toQuery(createContext());
+        assertThat(shouldClauses(boolQuery), is(4));
+
+        qb.defaultOperator(Operator.AND);
+        boolQuery = (BooleanQuery) qb.toQuery(createContext());
+        assertThat(shouldClauses(boolQuery), is(0));
+
+        qb.defaultOperator(Operator.OR);
+        boolQuery = (BooleanQuery) qb.toQuery(createContext());
+        assertThat(shouldClauses(boolQuery), is(4));
+    }
+
+    @Test
+    public void testValidation() {
+        SimpleQueryStringBuilder qb = createTestQueryBuilder();
+        assertNull(qb.validate());
+    }
+
+    @Test
+    public void testNullQueryTextGeneratesException() {
+        SimpleQueryStringBuilder builder = new SimpleQueryStringBuilder(null);
+        QueryValidationException exception = builder.validate();
+        assertThat(exception, notNullValue());
+    }
+
+    @Test
+    public void testHandlingDefaults() throws IOException {
+        SimpleQueryStringBuilder qb = createTestQueryBuilder();
+        qb.analyzer(null);
+        qb.minimumShouldMatch(null);
+        qb.queryName(null);
+        assertEquals(qb.toQuery(createContext()), createExpectedQuery(qb, createContext()));
+    }
+
+    
+    @Test(expected = IllegalArgumentException.class)
+    public void testFieldCannotBeNull() {
+        SimpleQueryStringBuilder qb = createTestQueryBuilder();
+        qb.field(null);
+    }
+    @Test(expected = IllegalArgumentException.class)
+    public void testFieldCannotBeNullAndWeighted() {
+        SimpleQueryStringBuilder qb = createTestQueryBuilder();
+        qb.field(null, 1.0f);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testFieldCannotBeEmpty() {
+        SimpleQueryStringBuilder qb = createTestQueryBuilder();
+        qb.field("");
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testFieldCannotBeEmptyAndWeighted() {
+        SimpleQueryStringBuilder qb = createTestQueryBuilder();
+        qb.field("", 1.0f);
+    }
+
+    /**
+     * The following should fail fast - never silently set the map containing
+     * fields and weights to null but refuse to accept null instead.
+     * */
+    @Test(expected = NullPointerException.class)
```

Comment:
Nope - throws NPE (implictly as it doesn't check the collection for being non null prior to accessing it).  In the SimpleQueryStringParser this isn't null when the method is called. So the only way is a JavaAPI user explicitly calling this method with null as an argument. I think in this case throwing NPE is more helpful/clearer for the downstream user.